src/main/java/org/opentripplanner/middleware/trip_monitor/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,176 @@
+        for (TripMonitorNotification notification : notifications) {
+            body.append(notification.body);
+        }
+        // FIXME: Change log level
+        LOG.info("Sending notification '{}' to user {}", subject, trip.userId);
+        // FIXME: This needs to be an enum.
+        switch (otpUser.notificationChannel.toLowerCase()) {
+            case "sms":
+                NotificationUtils.sendSMS(otpUser.phoneNumber, body.toString());
+                break;
+            case "email":
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            case "all":
+                NotificationUtils.sendSMS(otpUser.phoneNumber, body.toString());
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void checkTripForNewAlerts(MonitoredTrip trip, Itinerary itinerary) {
+        if (!trip.notifyOnAlert) {
+            LOG.debug("Notify on alert is disabled for trip {}. Skipping check.", trip.id);
+            return;
+        }
+        // FIXME: Need to somehow determine how to get last itinerary from responses in journey state.
+        Itinerary lastItinerary = trip.lastItinerary();
+        List<LocalizedAlert> newAlerts = itinerary.getAlerts();
+        List<LocalizedAlert> previousAlerts = lastItinerary.getAlerts();
+        // If journey state is already tracking alerts from previous checks, see if they have changed.
+        if (previousAlerts.size() > 0) {
+            // If there are new alerts, notify user.
+            List<LocalizedAlert> unseenAlerts = new ArrayList<>();
+            for (LocalizedAlert newAlert : newAlerts) {
+                boolean alertIsNew = true;
+                for (LocalizedAlert previousAlert : previousAlerts) {
+                    if (newAlert.equals(previousAlert)) {
+                        alertIsNew = false;
+                        break;
+                    }
+                }
+                if (alertIsNew) unseenAlerts.add(newAlert);
+            }
+            if (unseenAlerts.size() > 0) {
+                // FIXME: May need to update the user about alerts that have dropped off as well.
+                enqueueNotification(TripMonitorNotification.createAlertNotification(unseenAlerts));
+            }
+        } else {
+            // If there are new alerts present in the response, create notification.
+            if (newAlerts.size() > 0) {
+                enqueueNotification(TripMonitorNotification.createAlertNotification(newAlerts));
+            }
+        }
+    }
+
+    private void enqueueNotification(TripMonitorNotification tripMonitorNotification) {
+        notifications.add(tripMonitorNotification);
+    }
+
+    private boolean shouldSkipMonitoredTripCheck(MonitoredTrip trip) {
+        LocalDate now = LocalDate.now();
```

Comment:
This LocalDate is dependent on the system timezone. To future proof this, I think the most accurate way of doing this is to do a timezone lookup of the coordinates of the target trip end (the origin if the trip is a depart at trip, or the destination if the trip is an arrive by trip).

Reply:
This is an MVP, so I'm just going to add your comment in a FIXME.